mantidproject/mantid
# Store masking information in class DetectorInfo

This PR moves the storage location of mask flags for detectors from `ParameterMap` into `Beamline::DetectorInfo`. This is part of Instrument-2.0.

Masking data is intercepted when loading and added to `DetectorInfo` instead of `ParameterMap`. Masking data is injected into `ParameterMap` before saving. This ensures **backward and forward compatibility** for files (in particular processed Nexus).

For compatibility with legacy Python scripts a wrapper for `IDetector::isMasked` is provided. As a consequence docstests that are using this method are unchanged for now.


**To test:**



- Thorough code review.
- Manually check backward and forward compatibility of Nexus files: Have two Mantid builds (this branch and maybe a released version such as v3.8), in each of them mask detectors in a workspace, save as Nexus, try to load the file in the respective other build. Check that masking is loaded and looks the same.

Fixes #18290.
Fixes #18228.
Fixes #18319.
Fixes #18325.
Fixes #18351.
Fixes #18317.
Fixes #18354.

With the Python wrappers this is an internal change, so does not need to be in the release notes. We might observe performance improvements in a few cases, if so, those can be added later.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
